Addons/tsgExoticQuadrature.hpp

```diff
@@ -178,28 +197,77 @@ inline TasGrid::CustomTabulated getExoticQuadrature(const int n, const double sh
+    }
+    if (not std::all_of(weight_fn_vals.begin(), weight_fn_vals.end(), [shift](double x){return (x+shift)>=0.0;})) {
+        throw std::invalid_argument("ERROR: shift needs to be chosen so that weight_fn(x)+shift is nonnegative!");
+    }
+    for (size_t k=0; k<ref_points.size(); k++) {
+        shifted_weights[k] *= (weight_fn_vals[k] + shift);
+    }
+    return getShiftedExoticQuadrature(n, shift, weight_fn_vals, shifted_weights, ref_points, description, is_symmetric);
+}
+/*
+ * \ingroup TasmanianAddons
+ * \brief Constructs a CustomTabulated object containing the exotic quadrature nodes and quadrature weights for a given level \b n,
+ * scalar \b shift, a 1D function \b weight_fn() that returns the value of the weight function at its input, a positive integer
+ * \b nref that specifies how many points are used in estimating the entries of the Jacobi matrix, a string \b description,
+ * and an optional bool \b is_symmetric that should be set to true if the weight function is symmetric.
+ *
+ * The reference points and weights used for computing the entries of the Jacobi matrix are generated by the Gauss-Legendre rule
+ * with \b nref points and the weight function values are generated by applying \b weight_fn() to these reference points.
+ */
 inline TasGrid::CustomTabulated getExoticQuadrature(const int n, const double shift, std::function<double(double)> weight_fn,
                                                     const int nref, const char* description, const bool is_symmetric = false) {
     std::vector<double> shifted_weights, ref_points, weight_fn_vals(nref);
     TasGrid::OneDimensionalNodes::getGaussLegendre(nref, shifted_weights, ref_points);
     std::transform(ref_points.begin(), ref_points.end(), weight_fn_vals.begin(), weight_fn);
     if (not std::all_of(weight_fn_vals.begin(), weight_fn_vals.end(), [shift](double x){return (x+shift)>=0.0;})) {
-        throw std::invalid_argument("shift needs to be chosen so that weight_fn(x)+shift is nonnegative!");
+        throw std::invalid_argument("ERROR: shift needs to be chosen so that weight_fn(x)+shift is nonnegative!");
     }
     for (int k=0; k<nref; k++) {
         shifted_weights[k] *= (weight_fn_vals[k] + shift);
     }
-    return getExoticQuadrature(n, shift, weight_fn_vals, shifted_weights, ref_points, nref, description, is_symmetric);
+    return getShiftedExoticQuadrature(n, shift, weight_fn_vals, shifted_weights, ref_points, description, is_symmetric);
 }
 
-// If only the weight function is provided, the reference points are generated by the GL rule.
-
+/*
+ * \ingroup TasmanianAddons
+ * \brief Overloads of getExoticQuadrature() with some default values.
+ */
+inline TasGrid::CustomTabulated getExoticQuadrature(const int n, const double shift, TasGrid::TasmanianSparseGrid grid,
+                                                    const bool is_symmetric = false) {
+    return getExoticQuadrature(n, shift, grid, "Exotic quadrature", is_symmetric);
```

Comment:
`ExoticQuadrature` is not a good name, the name appears in error messages and status reports from Tasmanian. It is important that the user provides a meaningful name and it is good to always require is as an input.

Reply:
I removed the simplifying overloads for now.